docs/tutorials/algokit-template.md

```diff
@@ -0,0 +1,141 @@
+- `all`: The `all` command runs all the aforementioned bootstrap sub-commands in the current directory and its subdirectories. This command is a comprehensive way to ensure all project dependencies and environment variables are properly set up.
+
+### Predefined copier answers
+
+When initializing a new project, Copier can prompt the user for input, which is then passed to the template as variables. This is useful for customizing the new project based on user input.
+
+Example:
+
+```yaml
+# copier.yaml
+project_name:
+    type: str
+    help: What is the name of this project.
+    placeholder: "algorand-app"
+```
+This would prompt the user for the project name, and the input can then be used in the template using the Jinja syntax `{{ project_name }}`.
+
+#### Default Behaviors
+When creating an AlgoKit template, there are a few default behaviors that you can include to improve the development experience:
+
+- **Git**: If Git is installed on the user's system and the user's working directory is a Git repository, Copier will commit the newly created project as a new commit in the repository. This feature helps to maintain a clean version history for the project. If you wish to add a specific commit message for this action, you can specify a `commit_message` in the `_commit` option in your `copier.yaml` file.
+
+- **VSCode**: If the user has Visual Studio Code (VSCode) installed and the path to VSCode is added to their system's PATH, Copier can automatically open the newly created project in a new VSCode window. This is triggered by adding `vscode: true` in your `copier.yam` file.
+
+- **Bootstrap**: As previously mentioned, AlgoKit templates can include a bootstrap script that is automatically run after the project is initialized. This script can perform various setup tasks like installing dependencies or setting up databases. By default, if a `bootstrap` task is defined in the `copier.yaml`, it will be executed unless the user opts out during the prompt.
+
+By combining predefined Copier answers with these default behaviors, you can create a smooth, efficient, and intuitive initialization experience for the users of your template.
+## Recommendations
+
+- **Modularity**: Break your templates into modular components that can be combined in different ways.
+- **Documentation**: Include README files and comments in your templates to explain how they should be used.
+- **Versioning**: Use `.algokit.toml` to specify the minimum compatible version of AlgoKit.
+- **Testing**: Include test configurations and scripts in your templates to encourage testing best practices.
+- **Linting and Formatting**: Integrate linters and code formatters in your templates to ensure code quality.
+- **Algokit Principle**: for details on generic principles on designing templates refer to algo kit design principles
```

Comment:
Please also include link to https://github.com/algorandfoundation/algokit-cli/blob/main/docs/algokit.md#guiding-principles